lib/grpc_reflection/service/builder.ex

```diff
@@ -131,6 +133,33 @@ defmodule GrpcReflection.Service.Builder do
     end
   end
 
+  defp get_syntax(module) do
+    cond do
+      Keyword.has_key?(module.__info__(:functions), :__message_props__) ->
+        # this is a message type
+        case module.__message_props__().syntax do
+          :proto2 -> "proto2"
+          :proto3 -> "proto3"
+        end
+
+      Keyword.has_key?(module.__info__(:functions), :__rpc_calls__) ->
+        # this is a service definition, grab a message and recurse
+        module.__rpc_calls__()
+        |> Enum.find(fn
+          {_, _, _} -> true
+          _ -> false
+        end)
+        |> then(fn
+          nil -> raise "bad module, no biscuit"
+          {_, {req, _}, _} -> get_syntax(req)
+          {_, req, _} -> get_syntax(req)
+        end)
+
+      true ->
+        raise "Module #{inspect(module)} has neither rcp_calls nor __message_props__"
```

Comment:
for an empty service, we can't extract the syntax value. What about set `syntax` default to `proto2` rather than raise a message?

Reply:
defaulting seems safe enough, I'm not aware of a difference between proto2 and proto3 for an empty service